agent_testing/Bob.py

```diff
@@ -0,0 +1,66 @@
+"""Bob client to return route capable agent.
+
+Bob is intended to be run with he corresponding Alice server to
+demonstrate return routing.
+"""
+
+import hashlib
+import os
+from aries_staticagent import StaticConnection, Target, crypto, utils
+
+
+def main():
+    """Send a message and await the reply."""
+    their_vk, _ = crypto.create_keypair(seed=hashlib.sha256(b"server").digest())
+    conn = StaticConnection.from_seed(
+        hashlib.sha256(b"client").digest(),
+        Target(
+            their_vk=their_vk,
+            endpoint="http://localhost:{}".format(os.environ.get("PORT", 3000)),
+        ),
+    )
+
+    def list_addition():
+        """Creates an item to be added to the list."""
+
+        name = input("What do you want to name the" " item? : ")
+        features = input("What features are in this" " item? : ")
+
+        return {"Name": name, "Features": features}
+
+    def delete_item():
+        """Deletes an item from the given list"""
+
+        item_del = input("What is the Item Number of item you want to delete? : ")
+        return item_del
+
+    """Here we ask the client what they want to do."""
+    request = input("Hello! What do you need me to do? : ")
+
+    # The qualifier will determine the message type.
+    qualifier = ""
+
+    if "add" in request:
+        qualifier = "add"
+        request = list_addition()
+    elif "return" in request:
+        qualifier = "get"
+        request = "get_item"
+    elif "delete" in request:
+        qualifier = "delete"
+        request = delete_item()
+
+    reply = conn.send_and_await_returned(
+        {
+            "@type": "https://github.com/dbluhm/agent-testing/tree/main/docs/list/0.1/" + qualifier,
+            "~l10n": {"locale": "en"},
+            "sent_time": utils.timestamp(),
+            "content": request,
+        },
+        return_route="all",
+    )
+    print("Msg from conn:", reply and reply.pretty_print())
```

Comment:
Again, not going to make you make changes but just curious to hear your thoughts:
How would you change this program to let you perform multiple operations in one execution of the script?

Reply:
Operations in what sense? My thought in writing it this way was that if a single, more versatile operation could handle the message, then multiple operations wouldn't be necessary and that could cut down on runtime.

Reply:
To elaborate on what I mean, suppose I want to add an item to the list then delete the first item in the list. Right now, this is possible by running the program multiple times -- which is fine -- keeps things simple. I'm just probing how you would approach making it so I could do both the add and delete (or any number of operations) by just running the program once. :slightly_smiling_face:

Reply:
Oh got it! One of the problems I'm concerned about is if someone is unsure of the proper syntax to use when inputting requests. One way would be to just scan the incoming message for specific words like "add" and "get" and have them separated by commas, so that if someone wanted to do something like `add item2 feature2, delete 0` , then `item2` could be added while the index 0 item, `item1`, is being deleted. So multiple operations could occur in a single line. I could also write something like `What operations would you like to perform?` as the input option the client sees, and then they put in `add, get, delete` or however many operations they want to perform, their input is scanned for those specific words, and then the corresponding functions are carried out.